probatus/feature_elimination/feature_elimination.py

```diff
@@ -560,9 +589,21 @@ def fit(
                 for train_index, val_index in self.cv.split(current_X, self.y)
             )
 
-            shap_values = np.vstack([current_result[0] for current_result in results_per_fold])
-            scores_train = [current_result[1] for current_result in results_per_fold]
-            scores_val = [current_result[2] for current_result in results_per_fold]
+            if self.y.nunique() == 2 or is_regressor(current_model):
+                shap_values = np.vstack(
+                    [current_result[0] for current_result in results_per_fold]
+                )
+            else:  # multi-class case
+                shap_values = np.hstack(
```

Comment:
Why do we have hstacck here? How does the `current_result[0]` look for multiclass case

Reply:
Shape of `current_result[0]` is `(n_classes, n_rows, n_features)`. We're looking to stack the shap values in such a way that every feature gets a column and every fold gets an extra n_rows. hstack gives us a shape `(n_classes, n_folds * n_rows, n_features)`. Using vstack here would give us shape `(n_folds * n_classes, n_rows, n_features)`.